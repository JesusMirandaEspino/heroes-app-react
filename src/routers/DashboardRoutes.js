import React from 'react'
import { NavBar } from '../components/ui/NavBar';
import { Redirect, Route, Switch } from 'react-router';
import { MarvelScreen } from '../components/Marvel/MarvelScreen';
import { HeroesScreen } from '../components/Heroes/HeroesScreen';
import { DcScreen } from '../components/DC/DcScreen';

export const DashboardRoutes = () => {
    return (
        <div>

            <NavBar />


            <div className="container mt-5" >

                <Switch>

                    <Route  exact path="/marvel" component={ MarvelScreen } />
                    <Route  exact path="/hero/:heroeid" component={ HeroesScreen } />
                    <Route  exact path="/dc" component={ DcScreen } />

                    <Redirect to="/marvel"  />
                
                </Switch>

            </div>

            
        </div>
    )
}
