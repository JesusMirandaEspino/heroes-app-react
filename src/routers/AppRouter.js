import React, { useContext } from "react";
import {
    BrowserRouter as Router,
    Switch
} from "react-router-dom";
import  { DashboardRoutes } from './DashboardRoutes';


// Detenida import { NavBar } from '../components/ui/NavBar';
import { LoginScreen } from '../components/login/LoginScreen';
// Detenida import { MarvelScreen } from '../components/Marvel/MarvelScreen';

import { PrivateRouter } from '../routers/PrivateRouter';
import { PublicRoute } from '../routers/PublicRoute';
import { authContext } from '../auth/authContext';


export const AppRouter = () => {

    const { user } = useContext( authContext );

    return (
    <Router>
        <div  >
            <Switch>
                <PublicRoute  exact path="/login" component={ LoginScreen }  isAuthenticated={ user.logged } />
                <PrivateRouter  path="/" component={  DashboardRoutes  } isAuthenticated={ user.logged }  />
            </Switch>
        </div>

    </Router>
    )
}
