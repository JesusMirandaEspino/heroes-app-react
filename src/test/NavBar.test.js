import { NavBar } from '../components/ui/NavBar';
import React from 'react';
import { mount } from 'enzyme';
import { authContext } from '../auth/authContext';
import { AppRouter } from '../routers/AppRouter';
import { MemoryRouter } from 'react-router-dom';

describe( 'Pruebas con <NavBar />',  () => {

    const constexValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Jesus'
        }
    }

    const wrapper = mount(

            <authContext.Provider value={ constexValue } >
                <MemoryRouter>
                    <NavBar />
                </MemoryRouter>
            </authContext.Provider> 
        );

    test('Debe de mostrarlo correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( '.text-info' ).text().trim() ).toBe('Jesus');
    });
    


});