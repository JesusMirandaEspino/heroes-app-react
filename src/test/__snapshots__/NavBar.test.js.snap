// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <NavBar /> Debe de mostrarlo correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <p
        className="nav-item nav-link text-info"
      >
         
        Jesus
         
      </p>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
